### PR TITLE
Oauth2(GitHub) route created

### LINKED ISSUE #23: Set up GitHub OAuth

### PR DESCRIPTION
**Pull request by leventecsoba**
Note: its copied from an example code from the library docs, so its far from perfect

### CODE REVIEW

File: src/api/views/oauthView.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    return redirect(url_for('.profile'))
+
+@app.route("/profile", methods=["GET"])
+def profile():
+    github = OAuth2Session(client_id, token=session['oauth_token'])
+    user = github.get('https://api.github.com/user').json()
+    username = user['login']
+    email = user['email']
+
+    if(userController.get_user(username=username) = None):
```
Comment (ghost) [inline src/api/views/oauthView.py]: if(userController.get_user(username=username) == None):

**Review by ghost — CHANGES_REQUESTED** [review]
for this to work you need to add request_oauthlib to the pipfile
to do this run pipenv install request_oauthlib.

File: src/api/views/oauthView.py (hunk 1 above)
Comment (leventecsoba) [reply src/api/views/oauthView.py]: Thanks!

**Comment by leventecsoba** [comment]
> for this to work you need to add request_oauthlib to the pipfile
> to do this run pipenv install request_oauthlib.
The installation completes, but locking seems to fail...

File: src/api/views/oauthView.py (hunk 1 above)
Comment (ghost) [reply src/api/views/oauthView.py]: seems like I messed up.
You didnt need to add request-oauthlib.
I misssed the commit where you already added the relevant library to pipenv.
Please revert that change. after that we have to wait for the api keys to test it.

Comment (ghost) [reply src/api/views/oauthView.py]: can you also add the code to load the client_id and client secret from the .env file and give an error message if they have not been supplied
For reference look at src/api/models/database.py how the db_url is set and checked

Comment (ghost) [reply src/api/views/oauthView.py]: I fixed the stuffs

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
closes #23

**Comment by petak5** [comment]
I have tested the code and approve the changes.

**Pull request merged**